components/service/glean/src/test/java/mozilla/components/service/glean/storages/TimingDistributionsStorageEngineTest.kt

```diff
@@ -0,0 +1,167 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.glean.storages
+
+import android.content.Context
+import android.content.SharedPreferences
+import androidx.test.core.app.ApplicationProvider
+import kotlinx.coroutines.runBlocking
+import mozilla.components.service.glean.Lifetime
+import mozilla.components.service.glean.TimeUnit
+import mozilla.components.service.glean.TimingDistributionMetricType
+import mozilla.components.service.glean.resetGlean
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.ArgumentMatchers
+import org.mockito.Mockito
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
```

Comment:
Can we add test coverage for trying to accumulate negative values (check that errors are being reported), values that are too high (they should go in the last bucket)?